### ISSUE TITLE
Problem with Exploding Introspector + Automap + Synthetic Property

### ISSUE THREAD

**Issue by rsoffiatto** [dobrynya/nomin#28]
I'm trying to use the automap facility while having the coverage data collected by the JaCoCo agent, using the Exploding Introspector.
The JaCoCo agent instruments the tested code by adding a synthetic property named $jacocoData.
When the ExplodingIntrospector.java reads all the properties from the object being mapped, it reads the $jacocoData property and try to use it like the rest of the properties. This causes the NominException "Recursive mapping rule a = b causes infinite loop!"

As per the JaCoCo documentation, all the synthetic properties should be filtered. Indeed if I use a modified custom ExplodingIntrospector that filter all the synthetic properties everything works as expected.
My custom introspector only overrides the properties method, adding a findAll:

```groovy
    @Override
    Set<String> properties(Class<?> targetClass) {
        Set<String> properties = new HashSet()
        collectFields(targetClass).findAll { !it.synthetic }.each { properties.add(it.name) }
        return properties;
    }
```

**Comment by dobrynya** [comment]
Hi @rsoffiatto 
Does this fix work for you? Did you check that?

**Comment by dobrynya** [comment]
I have released nomin 1.1.4 with the required functionality. Please try it and let me know whether it works for you

**Comment by rsoffiatto** [comment]
I just tried. Yes, it is fixed. Thank you!
